test/config/integration/certs/certs.sh

```diff
@@ -2,32 +2,32 @@
 
 set -e
 
-# Uncomment the following lines if you want to regenerate the private keys.
-# openssl genrsa -out cakey.pem 1024
-# openssl genrsa -out serverkey.pem 1024
-# openssl genrsa -out clientkey.pem 1024
-# openssl genrsa -out upstreamcakey.pem 1024
-# openssl genrsa -out upstreamkey.pem 1024
+generate_ca() {
+  openssl genrsa -out cakey.pem 1024
+  openssl req -new -key $1cakey.pem -out $1cacert.csr -config $1cacert.cfg -batch -sha256
```

Comment:
There is way too much magic happening here when it comes to naming, IMHO.

Could you change those arguments to `$1key.pem`, `$1cert.cfg`, etc., and always invoke this function passing `$1`, i.e. `generate_ca ca` and `generate_ca upstreamca`?